cmd/hope/vm/image.go

```diff
@@ -35,31 +36,54 @@ var imageCmd = &cobra.Command{
 			return err
 		}
 
-		// Each image that's made will be copied to all hypervisors that
-		//   accept that image.
-		hypervisors, err := utils.GetHypervisors()
-		if err != nil {
-			return err
-		}
-
-		log.Debugf("Creating VM %s using %d hypervisors", vm.Name, len(vm.Hypervisors))
+		// Group hypervisors by engine
+		hypEngMap := map[string][]hypervisors.Hypervisor{}
```

Comment:
This hypervisor mapping bug fix could be extracted and merged independently. Would clear up this diff, and would allow this to get fixed sooner than however long the rest of this review takes.